sopc2dts/lib/SopcComponentLib.java

```diff
@@ -146,18 +146,23 @@ public void loadComponentLibsInWorkDir()
 				if(dirContents[i].endsWith(".xml")&&
 						dirContents[i].startsWith("sopc_components_"))
 				{
-					try {
-						Logger.logln("Loaded " + loadComponentLib(new InputSource(
-								new BufferedReader(new FileReader(dirContents[i]))))
-								 + " components from " + dirContents[i], LogLevel.DEBUG);
-					} catch (FileNotFoundException e) {
-						Logger.logException(e);
-					}
+					loadComponentLibFile(dirContents[i]);
 				}
 			}
 		}
 	}
 
+	public void loadComponentLibFile(String filename)
+	{
+		try {
+			Logger.logln("Loaded " + loadComponentLib(new InputSource(
+					new BufferedReader(new FileReader(filename))))
+					 + " components from " + filename, LogLevel.DEBUG);
+		} catch (FileNotFoundException e) {
+			Logger.logException(e);
+		}
+	}
+
 	public SopcComponentDescription getScdByClassName(String className)
```

Comment:
I like that refactured this to a new function instead of copy pasting.
Good Job.